Add terms from web fundamentals wiki

tannerwelsh opened the pull request:
Holy smokes there were a lot of terms in the web fundamentals wiki! https://github.com/codeunion/fundamentals-of-web-development/wiki/Glossary

A little bit of scripting later, and now they're here. None are quite as robust as the standard set by other terms, but its a start, no? And better to keep all the glossary terms in one place rather than spread around our repos.

tannerwelsh committed:
Add glossary entries from the web fundamentals repo

tannerwelsh committed:
Format all posts

- Remove extra whitespace
- Ensure newlines at end of files
- Ensure all posts have YAML front matter
- Minor copy edits

tannerwelsh commented:
@zspencer and/or @jfarmer whaddaya think? Should we add all these terms to the glossary?

zspencer commented on _posts/2014-01-01-application-framework.md:
```diff
@@ -0,0 +1,9 @@
+
+Often application frameworks are collections of many [libraries](/library) and have strong opinions about how to structure your application. Popular frameworks are often well documented, have tons of plugins or extensions, and take a while to learn.
+
+Not to be confused with [libraries](/library)
```
I think we already have /framework?

zspencer commented on _posts/2014-01-01-authorization-context.md:
```diff
@@ -0,0 +1,12 @@
+title: Authorization Context
+further-reading:
+---
+The Twitter API allows your application to work in two contexts: the application context and the user context.
```
Probably shouldn't reference twitter at this point, instead could talk about OAuth?

zspencer commented on _posts/2014-01-01-callback.md:
```diff
@@ -1,4 +1,5 @@
 ---
+title: Callback
```
Doesn't the title get capitalized automatically?

tannerwelsh replied:
Probably, I'm just trying to keep the YAML front matter consistent. As a rule, I try to keep any formatting consistent across all members of set so that someone who wants to create a new member can pick one at random to emulate and not diverge from the norm. In essence, everything is a "template".

zspencer commented on _posts/2014-01-01-dotenv.md:
```diff
@@ -0,0 +1,9 @@
+title: Dotenv
+further-reading:
+---
+We're using the `dotenv` [gem](/gem-library) to store sensitive information like your Twitter [consumer secret](/consumer-secret). This lets us use the line
```
This is also somewhat project/contextual that needs to be de-contextified.

zspencer commented on _posts/2014-01-01-heroku.md:
```diff
@@ -0,0 +1,5 @@
+title: Heroku
+further-reading:
+---
+Heroku is a service to [serve](/server-web) web applications for others to visit. It is free, but you may want to pay for upgrades until you get popular :smile:.
```
probably should be "pay for upgrades once you get popular"

zspencer commented on _posts/2014-01-01-json-javascript-object-notation.md:
```diff
@@ -0,0 +1,27 @@
+---
+JSON is a very common [data format](/data-format) used in [web services](/web-service-or-web-api). It is very human readable, and supported in many languages.
+
+The following is supermans hero information in a ruby hash and the JSON equivalent:
```
`Superman's`

zspencer commented on _posts/2014-01-01-params-sinatra.md:
```diff
@@ -0,0 +1,7 @@
+title: Params (Sinatra)
+further-reading:
+---
+Sinatra provides a hash of data from an http request in a helper method named `params`.
```
Should be `[HTTP request](/http-request)`

zspencer commented on _posts/2014-01-01-query-variables.md:
```diff
@@ -0,0 +1,9 @@
+
+They are part of a [URL](/url-uniform-reseource-locator)
+that comes after the `?`; and follow the format
+`name=value&other_name=other_value`.
```
Probably should mention that htey're key-value pairs as wel.

zspencer commented:
:+1: on a merge. There's a few nit-picks but nothing that should hold this up.

tannerwelsh committed:
Remove duplicate entry: Application Framework

tannerwelsh committed:
Don't reference Twitter API specifically

Instead, reference the more general "OAuth-enabled APIs"

tannerwelsh committed:
Remove context-dependent dotenv entry

tannerwelsh committed:
Minor text edits

tannerwelsh committed:
Prefix entries "request" and "response" with "http-" and update links

tannerwelsh committed:
Reference key-value pair in query variables

tannerwelsh committed:
Clean up entry titles and update links

- Make singular
- Use acronyms only, not in expanded form
- Add appropriate prefixes and suffixes

tannerwelsh committed:
Fix broken links in browser and update (sql)

tannerwelsh commented:
Thanks @zspencer! Addressed most of your points in subsequent commits.

Merging...

zspencer commented:
WOOO :dancers:
